pkg/policy/selectorcache.go

```diff
@@ -29,27 +29,51 @@ type scIdentity struct {
 
 // scIdentityCache is a cache of Identities keyed by the numeric identity
 type scIdentityCache struct {
-	ids map[identity.NumericIdentity]*scIdentity
+	ids         map[identity.NumericIdentity]*scIdentity
+	byNamespace map[string]map[*scIdentity]struct{}
 }
 
 func newScIdentityCache(ids identity.IdentityMap) scIdentityCache {
 	idCache := scIdentityCache{
-		ids: make(map[identity.NumericIdentity]*scIdentity, len(ids)),
+		ids:         make(map[identity.NumericIdentity]*scIdentity, len(ids)),
+		byNamespace: make(map[string]map[*scIdentity]struct{}, len(ids)),
 	}
 
 	for nid, lbls := range ids {
-		idCache.ids[nid] = newIdentity(nid, lbls)
+		idCache.insert(nid, lbls)
 	}
 
 	return idCache
 }
 
 func (c *scIdentityCache) insert(nid identity.NumericIdentity, lbls labels.LabelArray) {
-	c.ids[nid] = newIdentity(nid, lbls)
+	id := &scIdentity{
+		NID:       nid,
+		lbls:      lbls,
+		namespace: lbls.Get(labels.LabelSourceK8sKeyPrefix + k8sConst.PodNamespaceLabel),
```

Comment:
namespace is a useful predicate to push, but I could see us adding additional ones. Such as the presence of FQDN or CIDR sources?